app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt

```diff
@@ -20,20 +29,82 @@ class RoomReviewInfoFragment : Fragment(R.layout.fragment_room_review_info) {
     // Gets the shared view model
     private val viewModel by activityViewModels<ClassroomInfoViewModel>()
 
+    private lateinit var roomNoiseInfoTextView: TextView
+    private lateinit var noiseMeasureButton: ImageButton
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         viewModel.room.observe(viewLifecycleOwner) {
             view.findViewById<TextView>(R.id.roomIdInfo).text = it?.toString()
         }
-        viewModel.numReviews.observe(viewLifecycleOwner){
+        viewModel.numReviews.observe(viewLifecycleOwner) {
             view.findViewById<TextView>(R.id.roomNumReview).text =
                 getNumReviewString(requireContext(), it)
         }
-        viewModel.averageGrade.observe(viewLifecycleOwner){
+        viewModel.averageGrade.observe(viewLifecycleOwner) {
             view.findViewById<RatingBar>(R.id.roomRatingBar).rating = it.toFloat()
         }
+
+        roomNoiseInfoTextView = view.findViewById(R.id.roomNoiseInfoTextView)
+        viewModel.noiseData.observe(viewLifecycleOwner) {
+            displayRoomNoise(it)
+        }
+
+        // Record audio
+        val audioPermissionLauncher =
+            PermissionUtils.requestPermissionLauncher({ startAudio() }, this, requireContext())
+
+        noiseMeasureButton = view.findViewById(R.id.noiseMeasureButton)
+        noiseMeasureButton.setOnClickListener {
+            PermissionUtils.startPhoneFeature(
+                { startAudio() },
+                audioPermissionLauncher,
+                requireContext(),
+                Manifest.permission.RECORD_AUDIO
+            )
+        }
+
+    }
+
+    private fun displayRoomNoise(noiseData: Map<String, Int>) {
+        if (noiseData.isEmpty()) return
+
+        val sortedMap = noiseData.toSortedMap()
+        val mostRecentDate = sortedMap.lastKey()
```

Comment:
If we are just using the most recent recording, why keep all the older ones ?

Reply:
I wanted to also display a small list of recent measures but I did not find time to implement it. I can do this in the next sprint.